packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts

```diff
@@ -16,30 +18,30 @@ export enum OrgType {
 export async function getWorkspaceOrgType(
   defaultUsernameOrAlias?: string
 ): Promise<OrgType> {
-  if (!defaultUsernameOrAlias) {
-    const e = new Error();
-    e.name = 'NoDefaultusernameSet';
-    throw e;
-  }
-  const username = await OrgAuthInfo.getUsername(defaultUsernameOrAlias);
-  const isScratchOrg = await OrgAuthInfo.isAScratchOrg(username).catch(err =>
-    telemetryService.sendException(
-      'get_workspace_org_type_scratch_org',
-      err.message
-    )
-  );
-  return isScratchOrg ? OrgType.SourceTracked : OrgType.NonSourceTracked;
+  const connection = await WorkspaceContext.getInstance().getConnection();
+  const org: Org = await Org.create({ connection });
+  // Org.supportsSourceTracking() checks the org at this point in time.  This
+  // is important because a sandbox can be created without source tracking and
+  // then be refreshed to have source tracking.  So, in the case of sandboxes,
+  // we have to check the org's source tracking status in real time and can't
+  // rely on Org.tracksSource() because that property is populated at authorization-
```

Comment:
Did you look into whether or not we should be updating tracksSource directly rather than using this workaround?

Reply:
The Org.tracksSource method first checks the Auth file, and if the org has already been auth'd, it just grabs the property from there.  This method only makes a call to check the actual org if the Org has not yet been Auth'd.  So the core lib is doing some cacheing and trying to only make calls to the org when necessary.  It is useful as is for CLI so I don't think it should be changed for this PR.

The org.supportsSourceTracking() method always calls out to the org to get its source-tracking enabled state.  

In development I would see Org.tracksSource return false for scratch orgs sometimes in VSCE.   It seems built for transactional use in CLI, not long-running process use in VSCE.

Reply:
See above comment about switching to use tracksSource().